content/blog/2024/12/Deep-Learning-for-Gesture-Recognition-on-ESP32-S3-from-Training-to-Deployment/index.md

```diff
@@ -0,0 +1,384 @@
+    ├── SPI RAM config
+    │   ├── SPIRAM_MODE: Octal
+    │   ├── SPIRAM_SPEED: 40 MHz
+    │   └── Enable SPI RAM during startup
+    └── Allow .bss Segment Placement in PSRAM: Enabled
+```
+
+These configurations enable efficient utilization of external RAM for model weight storage and inference operations. The partition table configuration requires customization to accommodate the model data:
+
+```
+Partition Table
+└── Partition Table: Custom Partition Table CSV
+└── Custom Partition CSV File: partitions.csv
+```
+
+The custom partition layout ensures adequate space allocation for both the application and model data, facilitating efficient model loading during runtime.
+
+### Build System Integration
+
+The project's CMake configuration integrates the ESP-DL framework and establishes proper component relationships. The build system setup requires careful attention to include paths and dependencies:
+
+```cmake
+idf_component_register(
+    SRCS
+        "app_main.cpp"
+    INCLUDE_DIRS
+        "."
+        "model"
+    REQUIRES
+        esp-dl
+)
+```
+
+This configuration ensures proper compilation of the application components and correct linking with the ESP-DL framework. The integration of model files and headers follows a systematic approach through the build system.
+
+### Runtime Memory Management
+
+The implementation employs strategic memory allocation to optimize resource utilization during model inference. Memory monitoring capabilities are integrated into the system to ensure stable operation:
+
+```cpp
+size_t free_mem = heap_caps_get_free_size(MALLOC_CAP_SPIRAM);
+ESP_LOGI(TAG, "Available PSRAM: %u bytes", free_mem);
+```
+
+Model weights are stored in PSRAM while keeping critical runtime buffers in internal RAM for optimal performance. The system implements memory monitoring to prevent allocation failures and maintain stable operation during continuous inference.
+
+### Serial Communication Configuration
+
+The deployment setup includes specific serial communication parameters for reliable device interaction:
+
+```
+Component config
+└── ESP System Settings
+    └── Channel for console output
+        └── Default: UART0
+
+Component config
+└── Bluetooth
+    └── NimBLE Options
+        └── Host-controller Transport
+            └── Enable Uart Transport
+                └── Uart Hci Baud Rate: 115200
+```
+
+These settings ensure stable communication during both development and deployment phases. The baud rate selection balances reliable communication with flash programming speed.
+
+### Driver Installation and System Integration
+
+The deployment process requires proper USB driver installation for device communication. The CH340/CH341 driver installation process follows system-specific procedures, ensuring reliable device connectivity.
+
+## Experimental Results
+
+The gesture recognition system on ESP32-S3 showed strong performance across multiple metrics. Both quantitative and qualitative evaluations were conducted.
```

Comment:
If 'multiple metrics' were mentioned here, they should also be specified somewhere in the text, along with what can be considered as strong.